File: C#/Assets/Scripts/Room.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Room
{
    public Door[] doors { get; set; }
    public int[] coordinates { get; set; }

    public Room()
    {
        doors = new Door[4];
        coordinates = new int[] { 0, 0 };
    }

    public Room(Door[] newDoors, int[] newCoordinates)
    {
        doors = newDoors;
        coordinates = newCoordinates;
    }
}

File: C++/RecursiveDungeonGeneration/Room.cpp
#include "Room.h"

Room::Room() {
	for (int i = 0; i < sizeof(coordinates) / sizeof(int); i++) {
		coordinates[i] = -1;
	}

	for (int i = 0; i < sizeof(doors) / sizeof(Door); i++) {
		doors[i] = Door();
	}
}

Room::Room(int newCoordinates[2], Door newDoors[4]) {
	for (int i = 0; i < sizeof(coordinates) / sizeof(int); i++) {
		coordinates[i] = newCoordinates[i];
	}

	for (int i = 0; i < sizeof(doors) / sizeof(Door); i++) {
		doors[i] = newDoors[i];
	}
}